uwblueprint/ccbc
# deploy backend to Heroku

## Notion ticket link

[Deploy Backend](https://www.notion.so/uwblueprintexecs/Deploy-Backend-cd9dd364b4604ad4ac1dc737698c14e9)


## Implementation description
* deployed backend server and postgres db on Heroku (free tier pricing for now)
* managed firebase and nodemailer secrets through Heroku buildpack and config vars respectively
* added yarn build and start scripts to package.json to automate Heroku deployment
* made changes to sequelize options to allow server to connect with deployed db
* updated vault secrets
* updated cors origin list to allow client side deploys and deploy previews to make requests (see firebase reference [here](https://firebase.google.com/docs/hosting/manage-hosting-resources))



## Steps to test
1. Pull secrets `vault kv get -format=json kv/ccbc | python update_secret_files.py`
2. You should still be able to log in and sign up locally. Make sure node mailer in particular works as expected.
3. You can also try resetting your password to test nodemailer



## What should reviewers focus on?
* Testing!


## Checklist
